README.md

```diff
@@ -286,52 +292,169 @@ type Action = {
 
 #### Set Construction
 
-Actions are managed with an [LWW-Element-Set CRDT](https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type#LWW-Element-Set_(Last-Write-Wins-Element-Set)) which guarantees strong eventual consistency. Conceptually, there is a single **set** that stores all messages and conflicts are resolved by timestamp and lexicographical hash order. An addition is performed by constructing an `Action` message `a` where `active` is true, while a remove is performed by setting `active` to false. In both cases, the logic for merging the message into the set is: 
+Actions are managed with an [LWW-Element-Set CRDT](<https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type#LWW-Element-Set_(Last-Write-Wins-Element-Set)>) which guarantees strong eventual consistency. Conceptually, there is a single **set** that stores all messages and conflicts are resolved by timestamp and lexicographical hash order. An addition is performed by constructing an `Action` message `a` where `active` is true, while a remove is performed by setting `active` to false. In both cases, the logic for merging the message into the set is:
 
 1. If there is an action `x` in the set with the same values for `type`, `targetUri` and `account` as the incoming action `y`:
    - If `x > y`, discard `y`
    - If `x < y`, delete `x` and add `y` into the set
-2. Otherwise, add `y` into the set. 
-
+2. Otherwise, add `y` into the set.
 
 ## 4.3 Verifications
-*This section is still under development and will cover a CRDT for verified data types, including proof of ownership of an address or a specific NFTs*
+
+Verifications are bi-directional proofs of ownership between Farcaster accounts and external identities. Initially, Ethereum addresses are the only supported external identities, though other blockchains can be added later.
+
+Verifications must be opted in to by both the external identity (i.e. Ethereum address) and the Farcaster account. The bi-directional verification is implemented in two steps:
+
+1. Create a `VerificationClaim` object that contains the external address and the Farcaster account. Have the external identity sign a hash of the claim object, creating a directional proof from the external identity to the Farcaster account.
+2. Create a `VerificationAdd` message containing the external address and signature from step 1. Have the Farcaster account sign the message and share it with the network, wrapping and verifying the verification claim.
+
+In order to validate the `VerificationAdd` message, Farcaster hubs evaluate both the Farcaster signature as well as the external identity's signature. Initially, external signatures must implement [EIP 191 version 0x45](https://eips.ethereum.org/EIPS/eip-191) (i.e. Ethereum's `personal_sign`).
+
+Here are the schemas for `VerificationClaim` and `VerificationAdd`:
+
+```ts
+type VerificationClaim = {
+  externalAddressUri: string;
+  account: number;
+};
+
+type VerificationAdd = {
```

Comment:
To harden the schema, is it worth coming up with another type of proof just to test its extensibility? 

For example we could introduce a bi-directional "proof of web2 account" where you post a message to your Github or Twitter, connecting it to Farcaster.  This would be a bi-directional, but not self-authenticating proof.

Reply:
I mention other types at the beginning of the verification section (i.e. `Verifications can be used to prove ownership of Ethereum addresses, specific NFTs, other social media accounts, or even domains.`) I'd prefer not trying to support non self-authenticating proofs until thinking about it more, but what did you have in mind for the docs?